parts/k8s/kubernetesagentcustomdata.yml

```diff
@@ -130,6 +130,16 @@ write_files:
     }
 {{end}}
 
+- path: /etc/sysctl.d/10-oom-panic.conf
+  permissions: "0644"
+  owner: root
+  content: |
+    # These settings, and other Node settings are put in place by AKS. Do not attempt to change these values as any alterations will be removed by the service
+    # This setting disabled the linux OOM killer and sets panic on OOM to prevent partial node failures in out of memory conditions
+    vm.panic_on_oom = 1
+    # This setting instructs the kernel to reboot 30 seconds after a panic event to allow the node to automatically recover
+    kernel.panic = 30
```

Comment:
Just to play devil's advocate: why wait this long?

Reply:
We can go shorter if we want to. The recommended value for a watchdog reset (which is basically what this is) is 60 seconds. I thought that was a little too long for a node to be dead and not recovering, but I didn't want to swing too far either.

Reply:
Fair enough.